### PR TITLE
Add code

### PR DESCRIPTION
**Pull request by nof20**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# hello-kubernetes
+Example end-to-end build, test, and deploy pipelines for Kubernetes, using Google Cloud Build.
+
+# Instructions
+
+If you are new to Google Cloud Build, we recommend you start by visiting the [manage resources page](https://console.cloud.google.com/cloud-resource-manager) in the Cloud Console, [learn how to enable billing](https://cloud.google.com/billing/docs/how-to/modify-project), [enable the Cloud Build API](https://console.cloud.google.com/flows/enableapi?apiid=cloudbuild.googleapis.com), and [install the Cloud SDK](https://cloud.google.com/sdk/docs/).
```
Comment (bendory) [inline README.md]: suggest linking to the quick start for Go: https://cloud.google.com/cloud-build/docs/quickstart-go

File: README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+To try this out for yourself with a simple Go application which prints a greeting:
+
+* Fork this repo on GitHub by clicking the "Fork" icon in the top right-hand corner of the screen.
+* Open the [Google Cloud Build GitHub app](https://github.com/apps/google-cloud-build), click Install, and follow the setup instructions.  Select the `hello-kubernetes` repository you just created.
+* Back in your forked repo in GitHub, click to open `main_test.go`.  You will see some Go tests which make sure that the greeting matches the name provided.
+* Click the pencil icon in the top right-hand corner of the screen.
+* Edit one of the greetings so that the greeting does not match the name.
+* At the bottom of the screen, choose "Create a new branch and start a pull request".  Click "Propose file change".
+* On the page which opens, click "Create pull request".
+* You should see a message: "All checks have failed".  If you wish, you can see the build log by clicking the "Details" button and then selecting the link at the bottom of the screen which says "View more details on Google Cloud Build".
```
Comment (bendory) [inline README.md]: Suggest "...selecting the link in small print at the bottom of the screen..." -- I had trouble finding it.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+export PROJECT=$(gcloud info --format='value(config.project)')
+export PROJECT_NUMBER=$(gcloud projects describe $PROJECT --format 'value(projectNumber)')
+export CB_SA_EMAIL=$PROJECT_NUMBER@cloudbuild.gserviceaccount.com
+gcloud projects add-iam-policy-binding $PROJECT --member=serviceAccount:$CB_SA_EMAIL --role='roles/container.developer'  
+```
+
+Then:
+
+* Visit the [Build Triggers page](https://console.cloud.google.com/cloud-build/triggers) in the Cloud Console.
+* Click "Create Trigger".
```
Comment (bendory) [inline README.md]: It's actually "Add Trigger"

File: README.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+* Choose source GitHub and click Continue.  Follow the authorization flow to grant Google access to your code in GitHub.
+* Choose the `hello-kubernetes` repo, check the consent box, and click Continue.
+* Choose a name for your trigger, e.g. "Deploy to Kubernetes"
+* Under Branch (regex), enter `master`.
+* Under Build configuration, choose "cloudbuild.yaml", and enter filename `deploy.yaml`.
+* Click Save.
```
Comment (bendory) [inline README.md]: It's "Create Trigger"

File: README.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+* Now, return to GitHub, and click `main.go` to open the file.
```
Comment (bendory) [inline README.md]: This confused me -- have I abandoned my prior PR? Or am I still working in the same PR?

File: README.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+* Click the pencil icon on the top right-hand corner to edit.
+* On line 16, change the string inside `Greet("...")` to be your name.
```
Comment (bendory) [inline README.md]: Now I broke the test!
Suggest:

- up above, change the test to your name
- then to fix the broken test in the PR, change main.go to your name
- then when you come back here, come back to the PR, the test has passed, so merge into Master

File: README.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+* At the bottom of the page, ensure "Commit directly to the master branch" is selected, and click "Commit changes".
+* A container build, push, and deploy is now running in the background.  This normally takes about 20 seconds.  If you wish you can inspect logs in the [Build History](https://console.cloud.google.com/cloud-build/builds) page in the Cloud Console.
```
Comment (bendory) [inline README.md]: "Cloud Build"

File: main.go [hunk 8]
```diff
@@ -0,0 +1,22 @@
+//A very simple Go application.
```
Comment (bendory) [inline main.go]: nit: add a space after the '//' (here and in comments below too)

File: main.go [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+//Greet returns a pleasant greeting
+func Greet(name string) string {
+	return fmt.Sprintf("Hello %s!", name)
+}
+
+func main() {
+	greeting := Greet("change-me")
+	log.Printf(greeting)
+	for {
+		//Don't exit.
```
Comment (bendory) [inline main.go]: why the need not to exit? (otherwise I get distracted by this red herring)

File: README.md (hunk 1 above)
Comment (nof20) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (nof20) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (nof20) [reply README.md]: Looks like it actually changes depending on whether you have existing triggers or not!  Updated.

add-trigger

build-triggers

File: README.md (hunk 4 above)
Comment (nof20) [reply README.md]: Corrected, thx

File: README.md (hunk 5 above)
Comment (nof20) [reply README.md]: Made it clearer that the PR should be merged at the end of the first exercise.

File: README.md (hunk 7 above)
Comment (nof20) [reply README.md]: Good spot.  I simplified to "A build, push and deploy is now running..."

File: main.go (hunk 8 above)
Comment (nof20) [reply main.go]: Thanks... I feel like some code editor insisted I remove them at some point!

File: main.go (hunk 9 above)
Comment (nof20) [reply main.go]: Exiting immediately makes Kubernetes think we crashed, so it restarts, and after a handful of crash-loops it marks the container unhealthy.  I extended the comment to explain.

File: README.md (hunk 6 above)
Comment (nof20) [reply README.md]: Discussed in person, thanks!

**Pull request merged**